src/AbstractCollection.php

```diff
@@ -57,9 +62,162 @@ public function remove($element): bool
         if (($position = array_search($element, $this->data, true)) !== false) {
             unset($this->data[$position]);
 
+
             return true;
         }
 
         return false;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function column(string $propertyOrMethod): array
+    {
+        $temp = [];
+
+        foreach ($this->data as $item) {
+            $temp[] = $this->extractValue($item, $propertyOrMethod);
+        }
+
+        return $temp;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function first()
+    {
+        if (empty($this->data)) {
+            throw new OutOfBoundsException('Can\'t determine first item. Collection is empty');
+        }
+
+        reset($this->data);
+
+        return current($this->data);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function last()
+    {
+        if (empty($this->data)) {
+            throw new OutOfBoundsException('Can\'t determine last item. Collection is empty');
+        }
+
+        $item = end($this->data);
+        reset($this->data);
```

Comment:
`\end()` and `\reset()` ?

Reply:
Other function calls too, like `usort()` etc.

Reply:
Filed an issue for this: #46